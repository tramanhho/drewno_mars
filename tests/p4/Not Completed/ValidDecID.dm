//decl in differnt scopes

//global 
decl1 : int = 1;

//fn
decl2fn : (a : int, b: bool ) void { 
    decl2 : int = 2;
}

//global dec, fn assignment
decl3 : int;
decl3fn : (  ) void { 
    decl3 = 3;
    decl2fn ( 3, true);
}


