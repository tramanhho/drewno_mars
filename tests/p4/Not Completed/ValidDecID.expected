decl1{int} : int = 1;
decl2fn{(int, bool)->void} : (a{int} : int, b{bool} : bool) void  {
    decl2{int} : int = 2;
}
decl3{int} : int;
decl3fn{()->void} : () void  {
    decl3{int} = 3;
    decl2fn{(int, bool)->void}(3, true);
}